daft/runners/progress_bar.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+import os
+from typing import Any
+
+from tqdm.auto import tqdm
+
+from daft.execution.execution_step import PartitionTask
+
+
+class ProgressBar:
+    def __init__(self, use_ray_tqdm: bool, show_tasks_bar: bool = False, disable: bool = False) -> None:
+        self.use_ray_tqdm = use_ray_tqdm
+        self.show_tasks_bar = show_tasks_bar
+        self.tqdm_mod = tqdm
+        self.pbars: dict[int, tqdm] = dict()
```

Comment:
Initialize as `self.pbars = {-1: ...}`? We could also document here that `-1` represents the progress for "All tasks".

Reply:
we actually dont want to initialize the the progress bar until we get a task. Otherwise the total is 0 and it's in a weird state.